linux-4.20: fix cross compile for linux-headers package

marmeladema opened the pull request.

marmeladema commented:
This will help fixing #4641

Hoshpak commented:
Since the changes are non-trivial, it will take some time to review and test. If you don't hear from me within a few days, feel free to remind me.

marmeladema commented:
Basically it adds more file (Makefile, Kconfig, Kbuild, etc) in the linux-headers package for every architectures. Debian does the same thing.
It removes host-built helper binaries from the linux-headers package when cross compiling.
And add some missing directories needed to rebuilt those helper binaries on the target arch.

maxice8 commented on srcpkgs/linux4.20/template:
```diff
@@ -140,6 +140,13 @@ do_install() {
 
+	if [ "$CROSS_BUILD" ]; then
+		# Will transform x86_64 to x84-64 as reported by 'file' command
+		machine=$(echo $XBPS_MACHINE | tr '_' '-')
```
can be 

```
machine=${XBPS_MACHINE//_/-}
```

maxice8 replied:
Also, you declared this variable to use it only once ?

Hoshpak commented:
I successfully compiled a simple module with dkms on x86_64 with these changes so there seem to be no regressions. I completely agree with removing the x86_64 binaries, it just makes no sense to ship them on foreign architectures. However I'm very sceptical regarding the addition of 2406 Makefiles. Can't this be narrowed down to just the ones needed to achieve the result you want?

marmeladema commented:
@Hoshpak i squashed everything in one commit and fixed @maxice8 comment. I have also reverted my change to keep `arch/x86` to keep only `arch/x86/ras/Kconfig` instead.

marmeladema commented:
I confirm i successfully build an external module on an aarch64 board.
